core/src/main/java/org/vertexium/type/GeoLine.java

```diff
@@ -40,12 +41,12 @@ public GeoLine(GeoPoint start, GeoPoint end, String description) {
 
     @Override
     public boolean intersects(GeoShape geoShape) {
-        throw new VertexiumException("Not implemented for argument type " + geoShape.getClass().getName());
+        return GeoUtils.intersects(this, geoShape);
     }
 
     @Override
     public boolean within(GeoShape geoShape) {
-        throw new VertexiumException("Not implemented for argument type " + geoShape.getClass().getName());
+        return false;
```

Comment:
Is this true? A line can't be inside a shape?

Reply:
I was tripped up by this too, but [based on this implementation for `GeoRect`](I was tripped ) I interpret this to be asking if the `GeoShape` is inside a line. If we both thought this felt backwards, maybe we should flip the meaning of this method.

Reply:
Yeah maybe we should switch that. Very strange and doesn't read very well.

Reply:
Done